scripts/linters/pylint_extensions.py

```diff
@@ -1893,6 +1893,92 @@ def visit_compare(self, node):
                 self.add_message('inequality-with-none', node=node)
 
 
+class DisallowedFunctionsChecker(checkers.BaseChecker):
+    """Custom pylint checker for language specific general purpose
+    regex checks of functions calls to be removed or replaced.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    name = 'disallowed-function-calls'
+    priority = -1
+    msgs = {
+        'C0031': (
+            'Please remove the call to %s.',
+            'remove-disallowed-function-calls',
+            (
+                'Disallows usage of black-listed functions that '
+                'should be removed.'),
+        ),
+        'C0032': (
+            'Please replace the call to %s with %s.',
+            'replace-disallowed-function-calls',
+            (
+                'Disallows usage of black-listed functions that '
+                'should be replaced by allowed alternatives.'),
+        ),
+    }
+
+    options = (
+        (
+            'disallowed-functions-and-replacements',
+            {
+                'default': (),
+                'type': 'csv',
+                'metavar': '<comma separated list>',
+                'help': (
+                    'List of strings of disallowed function names. '
+                    'Strings should be either in the format \'A=>B\', '
+                    'where A is the disallowed function and B is the '
+                    'replacement or in the format \'A\', which signifies '
+                    'that A should just be removed.')
+            }
+        ),)
+
+    def _parse_config_entry(self, entry):
```

Comment:
Maybe change this to _fetch_disallowed_functions_and_replacements? (And also make this function independent of `entry` param as it can independently fetch config through self*)

Reply:
Will it be possible to parse this while initializing the checker and attach this to the object/self? So that we don't have to parse the config n times for n function calls in the module/file*

Reply:
I was able to make a change such that it should only fetch once, but was not able to do that at initialization time. Left an explanation in the code.

Reply:
Done with the `open` function!